### PR TITLE
[FLOC-4506] Remove the obsolete container API tests 

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4506

Also removed container agent related flocker-diagnostics tests and various ancillary code which appears not to be used elsewhere.

### CODE REVIEW

File: flocker/acceptance/testtools.py [hunk 1]
```diff
@@ -876,18 +747,6 @@ def cleanup_all_containers(_):
                 for container in client.containers():
                     client.remove_container(container["Id"], force=True)
```
Comment (wallnerryan) [inline flocker/acceptance/testtools.py]: is this calling the container API or docker?

File: flocker/acceptance/testtools.py [hunk 2]
```diff
@@ -954,7 +813,6 @@ def detach_and_destroy_volume(volume):
             )
 
         d = DeferredContext(cleanup_leases())
-        d.addCallback(cleanup_flocker_containers)
         if remove_foreign_containers:
             d.addCallback(cleanup_all_containers)
```
Comment (wallnerryan) [inline flocker/acceptance/testtools.py]: same, does this call docker or the container api?

File: flocker/acceptance/testtools.py [hunk 3]
```diff
@@ -1166,155 +1010,6 @@ def configure_container_agent(node):
...
-        d.addCallback(lambda _: loop_until(
-            reactor, lambda: is_process_running(
-                node, b'flocker-dataset-agent')))
-    # Hide the value in the callback as it could come from
-    # different places and shouldn't be used.
-    d.addCallback(lambda _: None)
-    return d
-
-
-def is_process_running(node, name):
```
Comment (wallnerryan) [inline flocker/acceptance/testtools.py]: this doesnt seem specific to container-agent removal, intended to be removed?

File: flocker/acceptance/testtools.py (hunk 2 above)
Comment (wallrj) [inline flocker/acceptance/testtools.py]: Yep, same thing. It calls the Docker API.

File: flocker/acceptance/testtools.py (hunk 3 above)
Comment (wallrj) [inline flocker/acceptance/testtools.py]: That function is no longer used inside Flocker so I decided to remove it.
Same for a bunch of other functions.
It seemed better than leaving them lying around and future maintainers having to think about why the function exists.

**Comment by wallrj** [comment]
I replaced this with #2917 so that I can merge into a remove container API feature branch.

**Pull request closed without merging**